lib/react_client/react_interop.dart

```diff
@@ -127,17 +128,156 @@ class JsRef {
+///   @override
+///   render() {
+///     final propsToForward = {...props}..remove('forwardedRef');
+///
+///     // Assign the custom prop `forwardedRef` as a ref on the component passed in via `props.component`
+///     return props['component']({...propsToForward, 'ref': props['forwardedRef']}, props['children']);
+///   }
+/// }
+/// final _logPropsHoc = react.registerComponent2(() => _LogProps());
+///
+/// final LogProps = react.forwardRef((props, ref) {
+///   // Note the second param "ref" provided by react.forwardRef.
+///   // We can pass it along to LogProps as a regular prop, e.g. "forwardedRef"
+///   // And it can then be attached to the Component.
+///   return _logPropsHoc({...props, 'forwardedRef': ref});
+///   // Optional: Make the displayName more useful for the React dev tools.
+///   // See: https://reactjs.org/docs/forwarding-refs.html#displaying-a-custom-name-in-devtools
+/// }, displayName: 'LogProps(${_logPropsHoc.type.displayName})');
+///
+/// // ---------- Component Consumption ----------
+/// void main() {
+///   setClientConfiguration();
+///   final ref = react.createRef<Element>();
+///
+///   react_dom.render(LogProps({'component': FancyButton, 'ref': ref}),
+///       querySelector('#idOfSomeNodeInTheDom'));
+///
+///   // You can still get a ref directly to the DOM button:
+///   final buttonNode = ref.current;
+/// }
+/// ```
 /// See: <https://reactjs.org/docs/forwarding-refs.html>.
-ReactJsComponentFactoryProxy forwardRef(Function(Map props, Ref ref) wrapperFunction) {
-  var hoc = React.forwardRef(allowInterop((JsMap props, JsRef ref) {
+ReactJsComponentFactoryProxy forwardRef(Function(Map props, Ref ref) wrapperFunction,
+    {String displayName = 'Anonymous'}) {
+  final wrappedComponent = allowInterop((JsMap props, JsRef ref) {
     final dartProps = JsBackedMap.backedBy(props);
     final dartRef = Ref.fromJs(ref);
     return wrapperFunction(dartProps, dartRef);
-  }));
+  });
+  defineProperty(wrappedComponent, 'displayName', jsify({'value': displayName}));
+
+  var hoc = React.forwardRef(wrappedComponent);
 
   return new ReactJsComponentFactoryProxy(hoc, shouldConvertDomProps: false);
 }
 
+/// Shorthand convenience function to wrap a JS component that utilizes `forwardRef`.
+///
+/// __Example:__
+///
+/// ```dart
+/// // Assuming that your app includes the JS file containing the MaterialUI components...
+/// @JS()
+/// class MaterialUI {
+///   external static ReactClass get IconButton;
+/// }
+///
+/// /// This makes it so that the `ref` set on the Dart `IconButton` component
+/// /// gets passed all the way to the root node rendered by the JS `MaterialUI.IconButton` component.
+/// final IconButton = forwardRefToJsComponent(MaterialUI.IconButton);
+///
+/// void main() {
+///   setClientConfiguration();
+///   final iconButtonRootNodeRef = react.createRef<Element>();
+///
+///   react_dom.render(IconButton({'ref': iconButtonRootNodeRef}, someIcon), querySelector('#idOfSomeNodeInTheDom'));
```

Comment:
I don't think this is the best example, because this example would function the same without the `forwardRefToJsComponent` call. Could we show it with the `additionalRefPropKeys` arg?